### DISCUSSION TITLE
Future: dfx identity list - see principals

### DISCUSSION THREAD

**Discussion by infu** [dfinity/sdk#3945] [Feature Requests]
The command `dfx identity list` could be improved slightly

Current output:
identity1
identity2
identity3
identity4
identity5

Requested output (Table containing the principals of each identity. Perhaps even cycle balances):

identity1    |    principal     |   wallet principal
identity2    |    principal     |   wallet principal
identity3    |    principal     |   wallet principal
...

**Comment by bitdivine** [comment]
I actually do this often enough that I have a script to do this. :-)

I prefer JSON output when programming but that could be an option.  E.g. there could be `dfx identity list --format=json` or similar.

**Reply by bitdivine** [reply]
I do have in mind that if `dfx` ever gets address book functionality, this should probably be consistent with the address book output.  I have a home-baked address book with commands such as `whois` (looks up a principal in my address book) and `whocontrols` (gets the principals controlling a canister and looks them all up in my address book).

**Comment by bitdivine** [comment]
The list of fields provided could be variable.  E.g. the `gh` cli supports a list of fields:
```
$ gh pr list --json
Specify one or more comma-separated fields for `--json`:
  additions
  assignees
  author
  autoMergeRequest
  baseRefName
...
$ gh pr list --json url,title
[
  {
    "title": "redacted",
    "url": "https://github.com/redacted"
  },
  ...
```
That way the command could start with offering just identity name and principal, for example, and add more values gradually.  No-longer supported fields can also be dropped in a graceful manner.
